packages/react-next/src/components/MessageBar/MessageBar.base.tsx

```diff
@@ -1,210 +1,138 @@
+  [MessageBarType.error]: 'ErrorBadge',
+  [MessageBarType.blocked]: 'Blocked2',
+  [MessageBarType.severeWarning]: 'Warning',
+  [MessageBarType.success]: 'Completed',
+};
+
+export const MessageBarBase = React.forwardRef<HTMLDivElement, IMessageBarProps>((props, ref) => {
+  const [expandSingleLine, { toggle: toggleExpandSingleLine }] = useBoolean(false);
+  const labelId = useId('MessageBar');
+
+  const {
+    actions,
+    className,
+    children,
+    overflowButtonAriaLabel,
+    dismissIconProps,
+    styles,
+    theme,
+    messageBarType = MessageBarType.info,
+    onDismiss = undefined,
+    isMultiline = true,
+    truncated,
+    dismissButtonAriaLabel,
+    messageBarIconProps,
+  } = props;
+
+  const nativeProps = getNativeProps<React.HTMLAttributes<HTMLSpanElement>>(props, htmlElementProperties, [
+    'className',
+  ]);
+
+  const classNames: { [key in keyof IMessageBarStyles]: string } = getClassNames(styles, {
+    theme: theme!,
+    messageBarType: messageBarType || MessageBarType.info,
+    onDismiss: onDismiss !== undefined,
+    actions: actions !== undefined,
+    truncated: truncated,
+    isMultiline: isMultiline,
+    expandSingleLine: expandSingleLine,
+    className,
+  });
+
+  const iconProps = { iconName: expandSingleLine ? 'DoubleChevronUp' : 'DoubleChevronDown' };
+
+  const getActionsDiv = React.useCallback((): JSX.Element | null => {
+    if (actions) {
+      return <div className={classNames.actions}>{actions}</div>;
+    }
+    return null;
+  }, [actions, classNames.actions]);
 
-  constructor(props: IMessageBarProps) {
-    super(props);
+  const getAnnouncementPriority = React.useCallback((): 'assertive' | 'polite' => {
+    switch (messageBarType) {
+      case MessageBarType.blocked:
+      case MessageBarType.error:
+      case MessageBarType.severeWarning:
+        return 'assertive';
+    }
+    return 'polite';
+  }, [messageBarType]);
 
-    initializeComponentRef(this);
-    this.state = {
-      labelId: getId('MessageBar'),
-      // eslint-disable-next-line react/no-unused-state
-      showContent: false,
-      expandSingleLine: false,
+  const getRegionProps = React.useCallback(() => {
+    const hasActions = !!actions || !!onDismiss;
+    const regionProps = {
+      'aria-describedby': labelId,
+      role: 'region',
     };
-  }
-
-  public render(): JSX.Element {
-    const { isMultiline } = this.props;
-
-    this._classNames = this._getClassNames();
-
-    return isMultiline ? this._renderMultiLine() : this._renderSingleLine();
-  }
-
-  private _getActionsDiv(): JSX.Element | null {
-    if (this.props.actions) {
-      return <div className={this._classNames.actions}>{this.props.actions}</div>;
-    }
-    return null;
-  }
+    return hasActions ? regionProps : {};
+  }, [actions, onDismiss, labelId]);
 
-  private _getDismissDiv(): JSX.Element | null {
-    const { onDismiss, dismissIconProps } = this.props;
+  const getDismissDiv = React.useMemo((): JSX.Element | null => {
```

Comment:
This one isn't expensive either, so just do `const dismissDiv = onDismiss ? <div....> : null`. Or in this case I'd probably prefer to just put it inline where it's used.